web/src/search_pill.ts

```diff
@@ -32,6 +56,36 @@ export function create_pills($pill_container: JQuery): SearchPillWidget {
     return pills;
 }
 
+function append_user_pill(
+    users: User[],
+    pill_widget: SearchPillWidget,
+    operator: string,
+    negated: boolean,
+): void {
+    const sign = negated ? "-" : "";
+    const search_string = sign + operator + ":" + users.map((user) => user.email).join(",");
+    const pill_data: SearchUserPill = {
+        type: "search_user",
+        operator,
+        // TODO(evy): make this "search_string" instead of "display_value"
+        display_value: search_string,
```

Comment:
It would be nice if we just call this `search_string` instead of `display_value`, but `display_value` is needed to hook into the generic input pill logic.

Reply:
The implementation here seems reasonable. Let's memorialize this thought as a `TODO` comment in the `SearchUserPill` type definition (or wherever feels most appropriate).

Reply:
`SearchUserPill` type definition makes sense to me, put it there :)